dashboard/utility_functions.py

```diff
@@ -0,0 +1,594 @@
+
+    Returns:
+        list: A list with review dates that the user selected
+    """
+    review_dates = st.sidebar.multiselect(
+        "Review Date:", options=df_releases["review_date"].dt.date.unique())
+    return review_dates
+
+
+def build_sidebar_genre(df_releases: DataFrame) -> list:
+    """
+    Build sidebar with dropdown menu options to select game genre
+
+    Args:
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A list with game genres that the user selected
+    """
+    df_releases = df_releases[df_releases["genre"].notna()]
+    selection = df_releases["genre"].unique()
+
+    genre = st.sidebar.multiselect("Genre:", options=selection)
+    return genre
+
+
+def build_sidebar_developer(df_releases: DataFrame) -> list:
+    """
+    Build sidebar with dropdown menu options to select game developer
+
+    Args:
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A list with game genres that the user selected
+    """
+    df_releases = df_releases[df_releases["developer_name"].notna()]
+    selection = df_releases["developer_name"].unique()
+
+    developer = st.sidebar.multiselect("Developer:", options=selection)
+    return developer
+
+
+def build_sidebar_publisher(df_releases: DataFrame) -> list:
+    """
+    Build sidebar with dropdown menu options to select game publisher
+
+    Args:
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A list with game genres that the user selected
+    """
+    df_releases = df_releases[df_releases["publisher_name"].notna()]
+    selection = df_releases["publisher_name"].unique()
+
+    publisher = st.sidebar.multiselect("Publisher:", options=selection)
+    return publisher
+
+
+def build_sidebar_platforms() -> list:
+    """
+    Build sidebar with platform selection options
+
+    Returns:
+        list: A list with selected platforms
+    """
+    platforms = ["mac", "windows", "linux"]
+    selected_platforms = st.sidebar.multiselect(
+        "Compatible Platforms:", options=platforms, default=platforms)
+    return selected_platforms
+
+
+def build_sidebar_price(df_releases: DataFrame) -> tuple:
+    """
+    Build sidebar with slider option to select range for review sentiment
+
+    Args:
+        df_releases (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A tuple with minimum and maximum sentiment that the user has selected
+    """
+    max_price = df_releases["price"].max()
+    min_price = df_releases["price"].min()
+    price = st.sidebar.slider(
+        "Price (£):", min_value=min_price, max_value=max_price,
+        value=(min_price, max_price), step=1.0)
+
+    return price
+
+
+def build_sidebar_sentiment(df_releases: DataFrame) -> tuple:
```

Comment:
fairly certain it would return a range and not tuple but maybe st formats them weirdly. Can you try printing `sentiment.__class__ ` before returning it?